File: src/main/java/pl/mroz/buddiesapi/domain/common/Location.java
package pl.mroz.buddiesapi.domain.common;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.ToString;
import pl.mroz.buddiesapi.infrastructure.database.location.LocationEntity;

import java.util.UUID;

@Getter
@EqualsAndHashCode
@ToString
@AllArgsConstructor
@Builder
public class Location {
    private UUID locationId;

    private String readableText;

    private double longitude;

    private double latitude;

    public static Location fromDb(LocationEntity entity) {
        return Location.builder()
                .locationId(entity.getLocationId())
                .readableText(entity.getReadableText())
                .longitude(entity.getLongitude())
                .latitude(entity.getLatitude())
                .build();
    }
}

File: src/main/kotlin/buddiesapi/domain/common/Location.kt
package buddiesapi.domain.common

import buddiesapi.infrastructure.database.location.LocationEntity
import java.util.UUID


class Location(
    val locationId: UUID?,
    val readableText: String,
    val longitude: Double,
    val latitude: Double
) {
    companion object {
        fun fromDb(db: LocationEntity): Location {
            return Location(db.locationId!!, db.readableText, db.longitude, db.latitude)
        }
    }
}
